src/librustc_typeck/check/compare_method.rs

```diff
@@ -301,10 +301,10 @@ fn compare_predicate_entailment<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
         debug!("compare_impl_method: trait_fty={:?}", trait_fty);
 
         let sub_result = infcx.sub_types(false, &cause, impl_fty, trait_fty)
-            .map(|InferOk { obligations, .. }| {
-                // FIXME(#32730) propagate obligations
-                assert!(obligations.is_empty());
-            });
+                              .map(|InferOk { obligations, .. }| {
+                                  fulfillment_cx.borrow_mut()
+                                                .register_predicate_obligations(infcx, obligations);
```

Comment:
IMO this should be using the equivalent `inh` instead... or not construct an `Inherited` at all.